### PR TITLE
Publish a benchmarks bundle along with each release

### PR DESCRIPTION
**Pull request by anandthakker**

### CODE REVIEW

File: bench/index.html [hunk 1]
```diff
@@ -12,16 +12,36 @@
...
     <div id="benchmarks"></div>
-    <script src="/bench/benchmarks_generated-v0.43.0.js"></script>
     <script>
-        // Unless the URL contains a no-master query parameter, include the
-        // current master-branch build of benchmarks.js for comparison
-        if (!/no-master/.test(window.location.search)) {
-            document.write(`<script src="https://s3.amazonaws.com/mapbox-gl-js/master/benchmarks.js"><` + `/script>`);
-        }
+        const params = {};
+        (window.location.search || '?').slice(1)
```
Comment (jfirebaugh) [inline bench/index.html]: How about using [`URLSearchParams`](https://stackoverflow.com/questions/901115/how-can-i-get-query-string-values-in-javascript/901144#12151322)?

File: bench/index.html [hunk 2]
```diff
@@ -12,16 +12,36 @@
...
+
+        Promise.resolve('compare' in params ?
+            params.compare.split(',').filter(Boolean) :
+            fetch('/package.json')
+            .then(response => response.json())
+            .then(pkg => [`v${pkg.version}`, 'master']))
+        .then(versions => {
+            const scripts = versions.map(v => `https://s3.amazonaws.com/mapbox-gl-js/${v}/benchmarks.js`)
+                .concat('/bench/benchmarks_generated.js', '/bench/benchmarks_view_generated.js');
+            let p = Promise.resolve();
```
Comment (jfirebaugh) [inline bench/index.html]: Unused.

File: bench/index.html [hunk 3]
```diff
@@ -12,16 +12,36 @@
...
+            return scripts.reduce((p, script) => p.then(() => loadScript(script)), Promise.resolve());
+        })
+        .catch(err => console.error(err))
+
+function loadScript(src) {
+    return new Promise(function (resolve, reject) {
+        var s;
```
Comment (jfirebaugh) [inline bench/index.html]: `const s = ...`

**Comment by jfirebaugh** [comment]
Should have mentioned: please also updated bench/README.md with the new query parameter format, and also document how to benchmark prior versions.

**Comment by anandthakker** [comment]
Ack, good call.

**Pull request merged**